### DISCUSSION TITLE
Welcome to engMQTTClient Discussions!

### DISCUSSION THREAD

**Discussion by gpbenton** [gpbenton/engMQTTClient#32] [General]
## 👋 Welcome!

 An interesting alternative to try, intead of raising issues

**Comment by Achronite** [comment]
Hi @gpbenton, I have a question for you, how did you find out what commands are supported by the eTRV?  I'm asking as I'm currently adding  MIHO069 thermostat support to my MQTT client https://github.com/Achronite/mqtt-energenie-ener314rt/issues/56 but the only commands I know *could* overwrite any manual 'button presses' on the device.

I've asked energenie too, but have not got anywhere yet...

**Comment by gpbenton** [comment]
I am not sure I ever tested what happens when the buttons are pressed on the device.  From memory I can only remember one function that is available on the etrv, and that boosts the the set temperature by a degree, and I have never used that, as its easier to get my phone out and set the temperature there :)

I would suspect that the temperature boost would apply until the eTRV received a new set temperature command, but as I say, I haven't tested it.

Its been a while since I've looked at my code for this project, but I seem to remember that I have a queue of commands for each trv, and the command at the head of the list gets sent in the next window when the eTRV sends a message.  I think I optimized the queue so that commands of the same type get overwritten by the late command, to avoid the queue getting long with  different set temperature commands.  I am a bit hazy, because its just been working and I haven't looked at it for a few years.

**Comment by Achronite** [comment]
I should have been clearer, the commands I'm after are for the Thermostat not the eTRV.  I'm looking for the command codes, such as THERMOSTAT_MODE=0xAA.  How did you work them all out?  Did you reuse other code or sniff the radio signals or something?

**Comment by gpbenton** [comment]
Ah, I don't have a thermostat, so I can't help you with that.
